api/tests/unit/segments/test_unit_segments_views.py

```diff
@@ -1041,21 +1154,20 @@ def test_update_segment_delete_existing_rule(project, client, segment, segment_r
                         "type": nested_rule.type,
                         "rules": [],
                         "conditions": [],
+                        "delete": True,
                     }
                 ],
                 "conditions": [],
-                "delete": True,
```

Comment:
This is something we need to discuss. The old implementation of sending `delete: true` allowed removing the top rule, but the new implementation of simply omitting them does not. I’ve updated this to align with the new approach of not allowing root-level rule deletion, since I don’t see a use case for it and the front end never deletes the root rule. I’m open to discussing changing the behavior back if we deem it necessary

Reply:
Seems logical to me.